cpp/tests/unit_tests/kernels/mlaPreprocessTest.cu

```diff
@@ -0,0 +1,708 @@
+        auto* k_pe_full_head_ptr = bufferCast<DataType>(*(this->k_pe_full_head_tensor));
+        auto* offset_ptr = bufferCast<int32_t>(*(this->compressed_offset_tensor));
+        auto* compressed_kv_cache_ptr = bufferCast<DataType>(*(this->compressed_kv_cache_tensor));
+        auto* cu_ctx_cached_kv_lens_ptr = bufferCast<int64_t>(*(this->cu_ctx_cached_kv_lens));
+        auto* cu_seq_lens_ptr = bufferCast<int64_t>(*(this->cu_seq_lens));
+        tensorrt_llm::kernels::KVBlockArray kv_cache(this->mNumRequests, this->mMaxBlockPerSeq, this->mTokensPerBlock,
+            sizeof(DataType) * 1 * (this->mLoraSize + this->mRopeSize), 0, 0, 0, 0, compressed_kv_cache_ptr, nullptr,
+            reinterpret_cast<tensorrt_llm::kernels::KVBlockArrayForContextFMHA::DataType*>(offset_ptr));
+        tensorrt_llm::kernels::invokeSetCompressedPagedKVKernel<DataType>(kv_cache, compressed_kv_ptr,
+            k_pe_full_head_ptr, this->mNumRequests, cu_ctx_cached_kv_lens_ptr, cu_seq_lens_ptr,
+            this->mMaxUncachedSeqLen, this->mLoraSize + this->mRopeSize, this->mStream->get());
+    }
+
+    void PerformKernel3Ref()
+    {
+        using tensorrt_llm::runtime::bufferCast;
+        auto* compressed_kv_ptr = bufferCast<DataType>(*(this->compressed_kv_tensor));
+        auto* k_pe_full_head_ptr = bufferCast<DataType>(*(this->k_pe_full_head_tensor));
+        auto* offset_ptr = bufferCast<int32_t>(*(this->compressed_offset_tensor));
+        auto* compressed_kv_cache_ptr = bufferCast<DataType>(*(this->compressed_kv_cache_tensor_ref));
+        auto* cu_ctx_cached_kv_lens_ptr = bufferCast<int64_t>(*(this->cu_ctx_cached_kv_lens));
+        auto* cu_seq_lens_ptr = bufferCast<int64_t>(*(this->cu_seq_lens));
+        tensorrt_llm::kernels::KVBlockArray kv_cache(this->mNumRequests, this->mMaxBlockPerSeq, this->mTokensPerBlock,
+            sizeof(DataType) * 1 * (this->mLoraSize + this->mRopeSize), 0, 0, 0, 0, compressed_kv_cache_ptr, nullptr,
+            reinterpret_cast<tensorrt_llm::kernels::KVBlockArrayForContextFMHA::DataType*>(offset_ptr));
+        // currently k_pe_head_num = 1
+        setCompressedPagedKvForMLAKernelRef(kv_cache, compressed_kv_ptr, k_pe_full_head_ptr, this->mNumRequests,
+            cu_ctx_cached_kv_lens_ptr, cu_seq_lens_ptr, 1, this->mLoraSize, this->mRopeSize);
+    }
+
+    template <typename T>
+    bool CheckEqual(T const* expected, T const* output, size_t size)
+    {
+        for (int i = 0; i < size; i++)
+        {
+            if (!almostEqual(expected[i], output[i], 1e-3, 1e-3))
+            {
+                TLLM_LOG_ERROR("Mismatch input value. Position of inputs: %d, expected value: %f, output value: %f", i,
+                    static_cast<float>(expected[i]), static_cast<float>(output[i]));
+                return false;
+            }
+        }
+        return true;
+    }
+};
+
+using MLATypes = ::testing::Types<half,
+#ifdef ENABLE_BF16
```

Comment:
This is not needed anymore. 
We shall always ENBLE_BF16. Volta support had been removed.